src/data_model.rs

```diff
@@ -0,0 +1,105 @@
+/// The size of a type.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[allow(missing_docs)]
+pub struct Size {
+    /// The size in bits of this type.
+    bits: u8,
+    /// The number of bits in a byte.
+    char_bit: u8,
+}
+
+impl Size {
+    /// Return the number of bits this `Size` represents.
+    pub fn bits(self) -> u8 {
+        self.bits
+    }
+
+    /// Return the number of bytes in a size.
+    ///
+    /// Depending on the target, a byte may not necessarily be 8 bits.
+    pub fn bytes(self) -> u8 {
```

Comment:
There are so many places that assume "byte" is "8-bit byte" these days that I'm worried this distinction is too subtle. What would you think about calling this method `c_chars` instead of `bytes` and said "Return the size in units of C `char`s (as C's `sizeof` does)"?